### PR TITLE
Sign using ACS

### PR DESCRIPTION
**Pull request by niik**
## Description

The desktop/desktop part of the changes needed to switch us over to signing our production/beta/test assets with ACS instead of our current certificate file.

## Release notes

Notes: no-notes

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -118,17 +123,22 @@ jobs:
...
+          # Replace ancient signtool in electron-winstall with one that supports ACS
+          Copy-Item -Path "C:\Program Files (x86)\Windows Kits\10\bin\10.0.22621.0\x64\*" -Include signtool.exe,signtool.exe.manifest,Microsoft.Windows.Build.Signing.mssign32.dll.manifest,mssign32.dll,Microsoft.Windows.Build.Signing.wintrust.dll.manifest,wintrust.dll,Microsoft.Windows.Build.Appx.AppxSip.dll.manifest,AppxSip.dll,Microsoft.Windows.Build.Appx.AppxPackaging.dll.manifest,AppxPackaging.dll,Microsoft.Windows.Build.Appx.OpcServices.dll.manifest,OpcServices.dll -Destination "node_modules\electron-winstaller\vendor" -Verbose
       - name: Package production app
         run: yarn package
         env:
           npm_config_arch: ${{ matrix.arch }}
-          WINDOWS_CERT_PASSWORD: ${{ secrets.WINDOWS_CERT_PASSWORD }}
+          AZURE_TENANT_ID: ${{ secrets.AZURE_CODE_SIGNING_TENANT_ID }}
+          AZURE_CLIENT_ID: ${{ secrets.AZURE_CODE_SIGNING_CLIENT_ID }}
+          AZURE_CLIENT_SECRET: ${{ secrets.AZURE_CODE_SIGNING_CLIENT_SECRET }}
```
Comment (tidy-dev) [inline .github/workflows/ci.yml:141]: How are these used? Is there something that looks for them in the environment.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -21,16 +21,21 @@ on:
       environment:
         type: string
         required: true
+      sign:
+        type: boolean
+        default: true
```
Comment (tidy-dev) [inline .github/workflows/ci.yml:26]: If this defaults true, will our regular pr CI builds attempt to sign?

Comment (tidy-dev) [reply .github/workflows/ci.yml:26]: I guess not as evidenced by this PR. 😄

File: .github/workflows/ci.yml (hunk 1 above)
Comment (niik) [reply .github/workflows/ci.yml:141]: Yes, these gets picked up by the Azure [EnvironmentCredential](https://learn.microsoft.com/en-us/dotnet/api/azure.identity.environmentcredential?view=azure-dotnet) provider. Azure tools follow an order of trying to acquire credentials that's documented here: https://learn.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet and the Environment provider is first up

**Review by tidy-dev — APPROVED** [review]
Approving to let this get moved in for beta. 

But, curious how the new env variables are used since there is no reference them here.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (niik) [reply .github/workflows/ci.yml:26]: Yeah, regular CI builds gets triggered by the `push` and `pull_request` triggers which doesn't define this input

Comment (tidy-dev) [reply .github/workflows/ci.yml:26]: Ah. Thanks for the explanation.

**Comment by MarinaM79** [comment]
https://github.com/desktop/desktop/pull/17648#discussion_r1377616587

**Pull request merged**